src/main/kotlin/gg/essential/universal/shader/UShader.kt

```diff
@@ -27,9 +29,21 @@ interface UShader {
     fun getSamplerUniform(name: String): SamplerUniform = getSamplerUniformOrNull(name) ?: throw NoSuchElementException(name)
 
     companion object {
+        @Deprecated(
+            "Use the overload which takes a vertex format to ensure proper operation on all versions.",
+            replaceWith = ReplaceWith("UShader.fromLegacyShader(vertSource, fragSource, blendState, vertexFormat)")
+        )
         fun fromLegacyShader(vertSource: String, fragSource: String, blendState: BlendState): UShader {
             //#if MC>=11700
-            //$$ return MCShader.fromLegacyShader(vertSource, fragSource, blendState)
+            //$$ return MCShader.fromLegacyShader(vertSource, fragSource, blendState, null)
+            //#else
+            return GlShader(vertSource, fragSource, blendState)
+            //#endif
+        }
+
+        fun fromLegacyShader(vertSource: String, fragSource: String, blendState: BlendState, vertexFormat: UGraphics.CommonVertexFormats): UShader {
```

Comment:
Might make sense to add another overload that takes a vanilla `VertexFormat`, so one can use it with custom vertex formats as well.

Reply:
I did it like this so that it's limited to vertex formats we know UC can process properly.